src/include/uapi/ipc/control.h

```diff
@@ -150,9 +150,13 @@ struct sof_ipc_ctrl_data {
 	/* control data - can either be appended or DMAed from host */
 	struct sof_ipc_host_buffer buffer;
 	uint32_t num_elems;	/**< in array elems or bytes for data type */
+	uint32_t total_elems;	/**< total size if message is sent in parts */
```

Comment:
we can probably use uint16_t for both total_elems and index. I would change total_elems to msg_parts.

Reply:
@lgirdwood total_elems is like total size in bytes or number of total array elems (as it is referring to union size). It is just reflecting the naming in num_elems, which is used in "normal small messaging". Many people seems to be confused by this .... maybe num_elems_total etc.?

Reply:
tbh, total_elems would be better as
```
uint32_t elems_remaining;
```
where this indicates how many elems/bytes are still to be sent. This also allows for a variable message size since we keep count of remaining data. So total size =  num_elems + remaining_elems when msg_index == 0

Reply:
@lgirdwood sorry for the delay. I was setting up my new up2 and debugging the bytes control user space setting for couple of days. I'll try this out, although it will make the fw side copy little bit more complicated as I can't use msg_index for offset (if variable size messages). elems_remaining -> elems_left ok with you? nicer shorter name....

Reply:
elems_left doesnt fit as well in English. remaining is obvious.